lib/identity_cache.rb

```diff
@@ -59,13 +59,13 @@ class << self
     # cache_has_many associations from the cache when never_set_inverse_association
     # is false. When set to true, it will only set the inverse cached association.
     mattr_accessor :never_set_inverse_association
-    self.never_set_inverse_association = version >= Gem::Version.new("0.5")
+    self.never_set_inverse_association = version >= Gem::Version.new("0.6")
 
     # Fetched records are not read-only and this could sometimes prevent IDC from
     # reflecting what's truly in the database when fetch_read_only_records is false.
     # When set to true, it will only return read-only records when cache is used.
     mattr_accessor :fetch_read_only_records
-    self.fetch_read_only_records = version >= Gem::Version.new("0.5")
+    self.fetch_read_only_records = version >= Gem::Version.new("0.6")
```

Comment:
these are breaking changes that we want to make for 0.5.  Don't just bump the number.  They are configurations so we can always use the old behaviour in Shopify until we can actually turn these on.